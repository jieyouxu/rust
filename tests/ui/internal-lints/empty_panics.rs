//@ compile-flags: -Z unstable-options

#![feature(rustc_private)]
#![forbid(rustc::empty_panics)]

fn main() {
    panic!()
    //~^ ERROR empty `panic!()`s do not provide any contextual information for debugging, please add some context or explanation
}

