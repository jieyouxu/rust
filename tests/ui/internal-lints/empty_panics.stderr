error: empty `panic!()`s do not provide any contextual information for debugging, please add some context or explanation
  --> $DIR/empty_panics.rs:7:5
   |
LL |     panic!()
   |     ^^^^^^^^ this `panic!()` does not have a message
   |
note: the lint level is defined here
  --> $DIR/empty_panics.rs:4:11
   |
LL | #![forbid(rustc::empty_panics)]
   |           ^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)

DEBUG rustc_lint::internal span=$SRC_DIR/std/src/panic.rs:LL:COL (#7), def_id=DefId(1:7505 ~ std[3461]::panicking::begin_panic), args=[&ReErased str]
error: aborting due to 1 previous error

